src/FlagProvider.tsx

```diff
@@ -107,7 +107,7 @@ const FlagProvider: FC<PropsWithChildren<IFlagProvider>> = ({
     () => ({
       on: ((event, callback, ctx) => client.current.on(event, callback, ctx)) as IFlagContextValue['on'],
       off: ((event, callback) => client.current.off(event, callback)) as IFlagContextValue['off'],
-      updateContext: async (context) => await client.current.updateContext(context),
+      updateContext: (context) => client.current.updateContext(context),
```

Comment:
It seems that the purpose of making it awaitable is to be able to have predictability on the tests. Maybe if you just return, I remember `async => await x` pattern is the same as `return x`, so maybe:
```suggestion
      updateContext: (context) => return client.current.updateContext(context),
```

Reply:
Actually, maybe simpler:
```suggestion
      updateContext: client.current.updateContext,
```